src/MainWindow.vala

```diff
@@ -818,33 +840,102 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
         var cursor_position = entry.cursor_position;
         entry.do_insert_text (input, -1, ref cursor_position);
         position += input.length;
-        entry.grab_focus ();
         entry.set_position (position);
     }
 
     private void remove_error () {
         infobar.revealed = false;
     }
 
-    private void replace_text (string new_text, int new_text_length, ref int position) {
-        var replacement_text = "";
-
-        switch (new_text) {
-            case ".":
-            case ",":
-                replacement_text = Posix.nl_langinfo (Posix.NLItem.RADIXCHAR);
-                break;
-            case "/":
-                replacement_text = "÷";
-                break;
-            case "*":
-                replacement_text = "×";
-                break;
+    private bool on_key_press (Gtk.EventControllerKey controller, uint keyval, uint keycode, Gdk.ModifierType mod_state) {
+        event_controller.forward (entry.get_delegate ());
+        switch (keyval) {
```

Comment:
Can you explain why you're doing this instead of leaving the entry sensitive? Why is this better than forwarding keypress events to be handled by the entry?

Reply:
Simple, because then I don't have to filter out all later or any special character the are not used in math functions, from the entry. 
Plus I had a look at the gnome calculator & this how they do it, But the difference is they are using a custom component & i'm not.

Reply:
> Can you explain why you're doing this instead of leaving the entry sensitive? Why is this better than forwarding keypress events to be handled by the entry?

Also read this https://github.com/elementary/calculator/issues/220

Reply:
> Can you explain why you're doing this instead of leaving the entry sensitive? Why is this better than forwarding keypress events to be handled by the entry?
Since entry currently allows the user enter letter & number & all special character, doing it this way means we don't have to do tons of error handling. basically entry is acting like the calculator display since Gtk.EventControllerKey is no taking over.